### DISCUSSION TITLE
Dreambooth extension very slow on MacOS M1

### DISCUSSION THREAD

**Discussion by wobba** [d8ahazard/sd_dreambooth_extension#733] [Q&A]
Hi,
SD webui works well and image generation is quick with 1.7it/s.

However the Dreambooth extension when generating class images is very slow. 150-300s/it. Looking at process usage I see it's CPU only. Any pointers on settings or code to change to improve speed under Dreambooth?

I assume it's running CPU only, so curious if it's possible to get GPU or neural network support.

**Comment by pallazola** [comment]
I'm also interested, so slow! txt2img maxes out the gpu, but dreambooth training doesn't touch it. I took a look at the python of the extension and saw alot of CUDA commands and no mps calls. It's beyond me to fork it and change it all to mps from CUDA. But I took a peek at the cloneofsimo/lora repo https://github.com/cloneofsimo/lora and it did seem to use the gpu when I train a few images
